packages/katello/katello/hostname-change.rb

```diff
@@ -345,6 +345,53 @@ def run
       STDOUT.puts "updating hostname in foreman installer scenarios"
       self.run_cmd("sed -i -e 's/#{@old_hostname}/#{@new_hostname}/g' #{scenarios_path}/*.yaml")
 
+      if @scenario_answers['foreman_proxy']['dns']
+        STDOUT.puts 'gathering DNS data...'
+        local_ip = '127.0.0.1'
+        zone = @scenario_answers['foreman_proxy']['dns_zone']
+        old_fqdn = @old_hostname
+        soa_admin_domain = "root.#{zone}"
+        zone_file = "/var/named/dynamic/db.#{zone}"
+        zone_file_data = File.read(zone_file).split("\n")
+        a_record = zone_file_data.find { |line| line.include?(' A ') }
+        existing_ip = a_record.split(' A ').last if a_record
+        serial_line = zone_file_data.find { |line| line.include?('Serial') }
```

Comment:
Is it needed to increase the serial when using nsupdate? Wouldn't it be safer to do a query for the SOA record and rely on the actual structure of the record rather than relying on a comment?

Reply:
@ekohl nsupdate does update the serial for you, except it doesn't seem to when you're updating the SOA record.